packages/component_code_gen/code_gen/generate_for_github_issue.py

```diff
@@ -72,7 +102,7 @@ def generate(issue_number, output_dir, verbose=False, tries=3):
 
     app_file_instructions = generate_app_file_prompt(
         "\n\n".join(extract_prompts(markdown)), app_file_content)
-    all_docs_urls = get_all_docs_urls(markdown)
+    all_docs_urls = list(set(get_all_docs_urls(markdown)))
```

Comment:
Did you still see cases of duplicate docs content even after this? This (making a set of all docs URLs) is what I was going to suggest re: the comment above. Maybe also strip the hashes at the end of the URL and get the set of stripped URLs? I noticed cases where Michael added the same URL with different anchor tags.

Reply:
Yes, some similar links contents were slightly different for some reason. Like one header another was different. So only creating a `set` did not resolve the issue. I was very often getting passed the token limit. This happens only in one-page markdown API docs like https://developer.ilovepdf.com/docs/api-reference

Reply:
Haven't tested stripping the hashes at the end of the URL, can try that

Reply:
Stripping the hashes works well!